src/main.spec.ts

```diff
@@ -34,99 +35,127 @@ test('calls global handler when any event is emitted', () => {
+    emitter.off(handler)
+    emitter.emit('hello')
 
-  expect(handler).toHaveBeenCalledTimes(2)
-})
+    expect(handler).toHaveBeenCalledTimes(2)
+  })
 
-test('stops calling the handler when returned off is called', () => {
-  const handler = vi.fn()
-  const emitter = createEmitter<{ hello: void }>()
+  test('given event without handler, removes all', () => {
+    const handlerA = vi.fn()
+    const handlerB = vi.fn()
+    const emitter = createEmitter<{ hello: void }>()
 
-  const off = emitter.on('hello', handler)
+    emitter.on('hello', handlerA)
+    emitter.on('hello', handlerB)
 
-  emitter.emit('hello')
-  emitter.emit('hello')
-  off()
-  emitter.emit('hello')
+    emitter.emit('hello')
+    emitter.emit('hello')
+    emitter.off('hello')
+    emitter.emit('hello')
 
-  expect(handler).toHaveBeenCalledTimes(2)
+    expect(handlerA).toHaveBeenCalledTimes(2)
+    expect(handlerB).toHaveBeenCalledTimes(2)
+  })
 })
 
-test('stops calling global handler when returned off is called', () => {
-  const handler = vi.fn()
-  const emitter = createEmitter<{ hello: void }>()
+describe('when returned off is called', () => {
+  test('stops calling the handler', () => {
+    const handler = vi.fn()
+    const emitter = createEmitter<{ hello: void }>()
 
-  const off = emitter.on(handler)
+    const off = emitter.on('hello', handler)
 
-  emitter.emit('hello')
-  emitter.emit('hello')
-  off()
-  emitter.emit('hello')
+    emitter.emit('hello')
+    emitter.emit('hello')
+    off()
+    emitter.emit('hello')
+
+    expect(handler).toHaveBeenCalledTimes(2)
+  })
+
+  test('stops calling global handler', () => {
+    const handler = vi.fn()
+    const emitter = createEmitter<{ hello: void }>()
 
-  expect(handler).toHaveBeenCalledTimes(2)
+    const off = emitter.on(handler)
+
+    emitter.emit('hello')
+    emitter.emit('hello')
+    off()
+    emitter.emit('hello')
+
+    expect(handler).toHaveBeenCalledTimes(2)
+  })
 })
 
-test('stops calling all handlers when clear is called', () => {
-  const handler = vi.fn()
-  const emitter = createEmitter<{ hello: void }>()
+describe('when clear is called', () => {
+  test('stops calling all handlers', () => {
+    const handler = vi.fn()
+    const emitter = createEmitter<{ hello: void }>()
 
-  emitter.on('hello', handler)
-  emitter.on(handler)
+    emitter.on('hello', handler)
+    emitter.on(handler)
 
-  emitter.emit('hello')
-  emitter.clear()
-  emitter.emit('hello')
+    emitter.emit('hello')
+    emitter.clear()
+    emitter.emit('hello')
 
-  expect(handler).toHaveBeenCalledTimes(2)
+    expect(handler).toHaveBeenCalledTimes(2)
+  })
 })
 
-test('event handler is called on multiple emitters when using useBroadcastChannel', async () => {
-  const handler = vi.fn()
-  const emitterA = createEmitter<{ hello: void }>({ broadcastChannel: 'my-channel' })
-  const emitterB = createEmitter<{ hello: void }>({ broadcastChannel: 'my-channel' })
+describe('when using useBroadcastChannel', () => {
```

Comment:
You mentioned this test was flakey. I never ran into that when testing this. But my guess is a broadcast channel has some time involved with it that's causing it to be slower sometimes. Maybe we can write a utility that resolves a promis only when a message was received or throws an error after like 10ms or something.

Reply:
Would that be in prod? Or are you suggesting a utility just for the test?

Reply:
just for the test. We just need to test that when broadcast channel is used the event makes its way across emitters. But I don't think that's a sync process which is why the `await timeout` is there. But that's probably not enough in this case. We probably need to await the actual broadcasted event.